hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/DecommissionStatusSubCommand.java

```diff
@@ -0,0 +1,153 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdds.scm.cli.datanode;
+
+import com.google.common.base.Strings;
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.cli.ScmSubcommand;
+import org.apache.hadoop.hdds.scm.client.ScmClient;
+import org.apache.hadoop.hdds.scm.container.ContainerID;
+import org.apache.hadoop.hdds.scm.container.ContainerInfo;
+import org.apache.hadoop.hdds.scm.container.ContainerNotFoundException;
+import org.apache.hadoop.hdds.scm.pipeline.Pipeline;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeOperationalState.DECOMMISSIONED;
+import static org.apache.hadoop.hdds.protocol.proto.HddsProtos.NodeOperationalState.DECOMMISSIONING;
+
+/**
+ * Handler to print decommissioning nodes status.
+ */
+@CommandLine.Command(
+    name = "decommissionStatus",
+    description = "Show decommission status for datanodes",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+
+public class DecommissionStatusSubCommand extends ScmSubcommand {
+
+  @CommandLine.Option(names = { "--decommissioned" },
+      description = "Show info of decommissioned nodes too",
+      defaultValue = "false")
+  private boolean decommissioned;
+
+  @CommandLine.Option(names = { "--id" },
+      description = "Show info by datanode UUID",
+      defaultValue = "")
+  private String uuid;
+
+  @Override
+  public void execute(ScmClient scmClient) throws IOException {
+    List<HddsProtos.Node> decommissioningNodes;
+
+    if (!Strings.isNullOrEmpty(uuid)) {
+      decommissioningNodes = scmClient.queryNode(DECOMMISSIONING, null,
+          HddsProtos.QueryScope.CLUSTER, "").stream().filter(p ->
+          p.getNodeID().getUuid().equals(uuid)).collect(Collectors.toList());
+      if (decommissioningNodes.isEmpty()) {
+        System.err.println("Datanode: " + uuid + " is not in DECOMMISSIONING");
+        return;
+      }
+    } else {
+      decommissioningNodes = scmClient.queryNode(DECOMMISSIONING, null,
+          HddsProtos.QueryScope.CLUSTER, "");
+      System.out.println("\nDecommission Status: DECOMMISSIONING - " +
+          decommissioningNodes.size() + " nodes");
+    }
+
+    Map<UUID, Map<HddsProtos.LifeCycleState, Long>> containerMap =
+        scmClient.getContainerMap();
+    Map<UUID, Integer> pipelineMap = scmClient.getPipelineMap();
+    List<Pipeline> currentPipelines = scmClient.listPipelines();
```

Comment:
About `listPipelines` - since we're only talking about decommission here, we can avoid getting data about all pipelines over the network, and instead only get the pipelines with decommissioning nodes.

Reply:
This too will be handled under [HDDS-9648](https://issues.apache.org/jira/browse/HDDS-9648)